back-end-projects/Explorer/src/main/java/com/github/ontio/service/impl/AddressServiceImpl.java

```
@@ -422,28 +422,34 @@ public ResponseBean queryAddressBalanceByAssetName(String address, String assetN
             String contractHash = map.get("contractHash");
 
             JSONArray balanceArray = sdk.getOpe8AssetBalance(address, contractHash);
-            balanceList.addAll(getBalancesListFromJson(balanceArray));
+            balanceList.addAll(getBalancesListFromJson(balanceArray, contractHash));
         }
 
         return balanceList;
     }
 
-    private String decodeAssetName(String hexAssetName) {
-        try {
-            return new String(Hex.decodeHex(hexAssetName.toCharArray()));
-        } catch (DecoderException e) {
-            return "";
+    private String getAssetName(String tokenId, String contractHash) {
+        Oep8 oep8 = new Oep8();
+
+        oep8.setTokenId(tokenId);
+        oep8.setContractHash(contractHash);
+
+        List<Oep8> oep8s = oep8Mapper.select(oep8);
+        if (oep8s.size() != 0) {
+            return oep8s.get(0).getSymbol();
         }
+
+        return "";
```

It's better to throw an exception here

---

Done.